File: tbvjgjfzm/2024-03/PGS_LV0_분수의덧셈.py
def solution(numer1, denom1, numer2, denom2):
    answer1 = 0
    answer2 = 0
    if (denom1 > denom2):
        if ((denom1 % denom2) == 0):
            answer1 = numer1 + denom1 // denom2 * numer2
            answer2 = denom1
        else : 
            answer1 = numer1 * denom2 + numer2 * denom1
            answer2 = denom1 * denom2
    elif(denom1 < denom2):
        if ((denom1 % denom2) == 0):
            answer1 = numer2 + denom2 // denom1 * numer1
            answer2 = denom2
        else : 
            answer1 = numer1 * denom2 + numer2 * denom1
            answer2 = denom1 * denom2
    elif(denom1 == denom2):
        answer1 = numer1 + numer2
        answer2 = denom1
    answer = [answer1, answer2]
    # 기약분수로 만들어야됨.
    answer3 = min(answer1, answer2) 
    for i in range(answer3, 1, -1): 
        if (((answer1 % i) == 0) and ((answer2 % i) == 0)): # i 자리에 answer3 넣음.
            answer = [answer1 // i, answer2 // i]
            break
    return answer

File: juju/2024-05/PGS_LV0_분수의_덧셈.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int numer1, int denom1, int numer2, int denom2) {
    // return 값은 malloc 등 동적 할당을 사용해주세요. 할당 길이는 상황에 맞게 변경해주세요.
    int* answer = (int*)malloc(sizeof(int)*2);
    int numer3 = (numer1 * denom2) + (numer2 * denom1);
    int denom3 = denom1 * denom2;
    int i = numer3;
    while(i > 0){
        if((numer3%i)==0 & (denom3%i)==0){
            numer3=numer3/i;
            denom3=denom3/i;
        }
        i--;
    }
    answer[0] = numer3;
    answer[1] = denom3;
    return answer;
}